### PR TITLE
Add writer style option to SettingsWriter pt1

### PR DESCRIPTION
**Pull request by opotowsky**
NOTE: the updates to the original PR on this topic requested by @keckler will be in a follow-up PR, namely making this argument available to `clone` / `writeInputs`.

## Description

As described in Issue #897, we would like a third option that preserves the original settings file even if a setting defined in it is a default value. Currently, `short` removes all settings that have the default value, and `full` prints out every possible setting to the file.

To accomplish that, I added a `medium` style option to `SettingsWriter` (which involved editing the handling of the strings of "acceptable" styles to constant strings as well as a helper function in the `caseSettings` module called `getSettingSetByUser` so the full list of user defined settings can be tracked.).

I made the `settingsWriteStyle` flag available to the `modify` EntryPoint, so one can alter the writing style from the command line.

The testing was a little difficult. `writeToYamlFile` method and `SettingsWriter` class aren't explicitly tested. I have the 3 writing style options tested, the new method (`getSettingsSetByUser`) in `caseSettings.py` is tested, and a sad attempt at adding this to the `modify` CLI testing. 

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/settings/caseSettings.py [hunk 1]
```diff
@@ -368,13 +368,63 @@ def writeToYamlFile(self, fName, style="short"):
...
+
+        Parameters
+        ----------
+        stream : file object
+            Write-only file stream
+
+        style : str
+            Writing style for settings file. Can be short, medium, or full.
+
+        settingsSetByUser : list
```
Comment (keckler) [inline armi/settings/caseSettings.py:417]: Specify as optional.

File: armi/settings/caseSettings.py [hunk 2]
```diff
@@ -368,13 +368,63 @@ def writeToYamlFile(self, fName, style="short"):
...
+        ----------
+        fPath : str
+            The absolute file path of the settings file
+
+        Returns
+        -------
+        userSettingsNames : list
+            The settings names read in from a yaml settings file
+        """
+        from ruamel.yaml import YAML
```
Comment (keckler) [inline armi/settings/caseSettings.py:394]: I am assuming there is some reason that this cannot be at the module level? If not, we should put it up there.

Comment (opotowsky) [reply armi/settings/caseSettings.py]: Yea there's no reason other than me not cleaning things up properly -- done!

File: armi/settings/settingsIO.py [hunk 3]
```diff
@@ -320,7 +324,14 @@ def _getSettingDataToWrite(self):
         for _settingName, settingObject in iter(
```
Comment (jakehader) [inline armi/settings/settingsIO.py:324]: Recommend removing the `_` from `_settingName` now that it being used.

**Review by jakehader — CHANGES_REQUESTED** [review]

**Review by jakehader — APPROVED** [review]

**Review by john-science — APPROVED** [review]
I love the test coverage!

**Pull request merged**